ext/liquid_c/tokenizer.c

```diff
@@ -114,6 +115,16 @@ void tokenizer_next(tokenizer_t *tokenizer, token_t *token)
     token->length = cursor - tokenizer->cursor;
     tokenizer->cursor += token->length;
     tokenizer->length -= token->length;
+
+    if (tokenizer->line_number) {
+        const char *cursor = token->str;
+        const char *end = token->str + token->length;
+        while (cursor < end) {
+            if (*cursor == '\n')
+                tokenizer->line_number++;
+            cursor++;
+        }
+    }
```

Comment:
It would involve doing a check at a number of places in this method (https://github.com/shopify/liquid-c/commit/4151be34ea3ebd46dc07d6e82152b3b9f37d299b) and would also add unnecessary overhead when line numbers aren't getting used.